native/src/main/java/io/ballerina/lib/aws/sqs/listener/Caller.java

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2025 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package io.ballerina.lib.aws.sqs.listener;
+
+import io.ballerina.lib.aws.sqs.client.NativeClientAdaptor;
+import io.ballerina.runtime.api.values.BError;
+import io.ballerina.runtime.api.values.BObject;
+import software.amazon.awssdk.services.sqs.SqsClient;
+import software.amazon.awssdk.services.sqs.model.DeleteMessageRequest;
+
+import static io.ballerina.lib.aws.sqs.CommonUtils.createError;
+
+/**
+ * Native implementation of the Ballerina AWS SQS Caller object.
+ * Handles message acknowledgment and negative acknowledgment operations.
+ */
+public final class Caller {
+
+    private static final String NATIVE_SQS_CLIENT = NativeClientAdaptor.NATIVE_SQS_CLIENT;
```

Comment:
Do we have to redefine this constant here as well ? Shall we have a common `Constant` class and move the common constants to that class ?